bait-files/src/main/webapp/submission.js

```diff
@@ -0,0 +1,248 @@
+    createOption("New Brunswick", province);
+    createOption("Newfoundland and Labrador", province);
+    createOption("Northwest Territories", province);
+    createOption("Nova Scotia", province);
+    createOption("Nunavut", province);
+    createOption("Ontario", province);
+    createOption("Prince Edward Island", province);
+    createOption("Quebec", province);
+    createOption("Saskatchewan", province);
+    createOption("Yukon", province);
+
+    provinceDiv.appendChild(province);
+    addressInputDiv.appendChild(provinceDiv);
+
+    //Postal Code
+    createTextInputElement("Postal", document.getElementById("address-input-div"));
+    
+}
+
+function createOption(optionValue, selectToAppend) {
+    let option = document.createElement("option");
+    option.setAttribute("value", optionValue)
+    let optionText = document.createTextNode(optionValue);
+    option.appendChild(optionText);
+    selectToAppend.appendChild(option);
+}
+
+function createTextInputElement(inputElement, divToAppend) {
+    let inputDiv = document.createElement("div");
+    inputDiv.setAttribute("id", (inputElement + "-div"));
+    let inputLabel = document.createElement('label');
+    inputLabel.setAttribute("for", (inputElement + "-input"));
+    let inputLabelText = document.createTextNode(inputElement + ": ");
+    inputLabel.appendChild(inputLabelText);
+    inputDiv.appendChild(inputLabel);
+    let input = document.createElement("input");
+    input.type = "text";
+    input.setAttribute("id", (inputElement + "-input"));
+    input.setAttribute("name", (inputElement + "-input"));
+    inputDiv.appendChild(input);
+    divToAppend.appendChild(inputDiv);
+}
+
+function clearDiv(divToClear) {
+  while (divToClear.firstChild) {
+    divToClear.removeChild(divToClear.lastChild);
+  }
+}
+
+function codeAddress(geocoder, map, address) {
+    clearMarkers();
+    geocoder.geocode({'address': address}, function(results, status) {
+        if (status === 'OK') {
+            map.setCenter(results[0].geometry.location);
+            addMarker(results[0].geometry.location);
+        } else {
+            alert('Geocode was not successful for the following reason: ' + status);
+        }
+    });
+}
+
+let countryInput = document.getElementById("country-input");
+countryInput.addEventListener('change', (event) => {
+    let addressInputDiv = document.getElementById("address-input-div");
+    clearDiv(addressInputDiv);
+    if (event.target.value == "US") {
+        loadAmericanAddress();
+    } else if (event.target.value == "CA"){
+        loadCanadianAddress();
+    }
+});
+
+function placeMarkerOnUserInputtedAddress() {
+    let country = document.getElementById("country-input").value;
+    if (country == "US") {
+        let addressLineOne = document.getElementById("Address Line 1-input").value;
+        let addressLineTwo = document.getElementById("Address Line 2-input").value;
+        let city = document.getElementById("City-input").value;
```

Comment:
again any class/id names should probably be lowercase with dashes instead of spaces

Reply:
Lowercased (with new function)